### DISCUSSION TITLE
bug: Unsupported Basic Auth

### DISCUSSION THREAD

**Discussion by Majorfi** [wevm/wagmi#2873] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.3.3

### Current Behavior

Right now, if you try to use a jsonRpcProvider with a http url with basic auth, an error will be triggered by fetch, because fetch does not support basic auth in url.

### Expected Behavior

This should transform the basic auth to `Authorization` header, like [explained here](https://btb.works/javascript-fetch-with-basic-auth/) for example.

### Steps To Reproduce

On the config, add a rpc url with basic auth

```ts
jsonRpcProvider({
	rpc: (chain): {http: string} => ({http: 'https://username@password:somethingsomething.com'})
})
```

### Anything else?

This can be solved by updating the [utils/configureChains.ts](https://github.com/wagmi-dev/wagmi/blob/13c970798b5a2b0e2a5d32ad338d15dfaa48f86f/packages/core/src/utils/configureChains.ts#L95C1-L103C9) file with something similar

```diff
const publicClient = createPublicClient({
    batch,
    chain: activeChain,
    transport: fallback(
-     chainHttpUrls.map((url) => http(url, { timeout: stallTimeout })),
+     chainHttpUrls.map((url: string) => {
+       const urlAsNodeURL = new URL(url);
+       let headers = {};
+       if (urlAsNodeURL.username && urlAsNodeURL.password) {
+           headers = {
+               'Authorization': `Basic ${btoa(urlAsNodeURL.username + ':' + urlAsNodeURL.password)}`
+           };
+           url = urlAsNodeURL.href.replace(`${urlAsNodeURL.username}:${urlAsNodeURL.password}@`, '');
+       }
+       return http(url, {timeout: stallTimeout, fetchOptions: {headers}});
+   }),
      { rank, retryCount, retryDelay },
    ),
    pollingInterval,
})
```

**Comment by jxom** [comment]
Good idea! Going to move this to discussions as it's not really a bug (more lack of feature), but we won't forget about it! Open to PR.
